py/builtinimport.c

```diff
@@ -491,6 +503,12 @@ STATIC mp_obj_t process_import_at_level(qstr full_mod_name, qstr level_mod_name,
     }
     #endif // MICROPY_MODULE_OVERRIDE_MAIN_IMPORT
 
+    // Make sure we remove the module from sys.modules if there was an
+    // exception while running its code.
+    nlr_jump_callback_node_unregister_module_t ctx;
+    ctx.full_mod_name = full_mod_name;
+    nlr_push_jump_callback(&ctx.callback, unregister_module_from_nlr_jump_callback);
```

Comment:
Should this be move earlier, in case the above `mp_obj_dict_store` fail?

Reply:
Yeah, it probably makes sense for the module to be removed if anything goes wrong during the rest of the `process_import_at_level` function after the module was registered.